src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -123,113 +111,34 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
             TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA),
             TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
 
-        // request the latest LLM key and endpoint
-        _AIKeyAndEndpointRequestedHandlers(nullptr, nullptr);
+        IResponse result;
+
+        // Make a copy of the prompt because we are switching threads
+        const auto promptCopy{ prompt };
+
+        // Start the progress ring
+        IsProgressRingActive(true);
 
-        // Use a flag for whether the response the user receives is an error message
-        // we pass this flag to _splitResponseAndAddToChatHelper so it can send the relevant telemetry event
-        // there is only one case downstream from here that sets this flag to false, so start with it being true
-        bool isError{ true };
-        hstring result{};
+        // Make sure we are on the background thread for the http request
+        co_await winrt::resume_background();
 
-        // If the AI key and endpoint is still empty, tell the user to fill them out in settings
-        if (_AIKey.empty() || _AIEndpoint.empty())
+        if (_llmProvider)
```

Comment:
Yea this does feel bodgy currently, but there's only one provider for now so it seems totally fine. When there are other providers, it seems like it'd make more sense to have separate errors for "you haven't set up any LLM backend" (generic, from the Extension palette itself) vs "you didn't set up an API key" (from the individual providers)